traversal/predicate/PredicateArgument.java

```diff
@@ -82,11 +89,16 @@ public boolean apply(PredicateOperator.Equality operator, AttributeVertex<?> ver
             public boolean apply(PredicateOperator.Equality operator, AttributeVertex<?> vertex, Boolean value) {
                 if (!vertex.valueType().comparableTo(Encoding.ValueType.BOOLEAN)) return false;
                 assert vertex.isBoolean();
-                return operator.apply(Predicate.compareBooleans(vertex.asBoolean().value(), value));
+                return apply(operator, vertex.asBoolean().value(), value);
+            }
+
+            @Override
+            public boolean apply(PredicateOperator.Equality operator, Boolean lhs, Boolean rhs) {
+                return operator.apply(valueType().comparator().compare(lhs, rhs));
```

Comment:
new API that allows us to compare a LHS and RHS of the same type using these `PredicateArgument` classes, rather than requiring that we use a vertex + value or a vertex + parametrised value.